src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -2235,7 +2235,7 @@ static int32_t WaitForSocketEventsInner(int32_t port, struct SocketEvent* buffer
         struct kevent evt = events[i];
         memset(&buffer[i], 0, sizeof(struct SocketEvent));
         buffer[i].Data = GetSocketEventData(evt.udata);
-        buffer[i].Events = GetSocketEvents(GetKeventFilter(evt.filter), GetKeventFlags(evt.flags));
+        buffer[i].Events = (int32_t)(GetSocketEvents(GetKeventFilter(evt.filter), GetKeventFlags(evt.flags)));
```

Comment:
why does this need a cast? Is this casting 64bit to 32bit or just signed/unsigned cast?

Reply:
SocketEvents Enum to int32_t cast